maven-core/src/main/java/org/apache/maven/bridge/MavenRepositorySystem.java

```diff
@@ -275,9 +288,14 @@ public void injectProxy(RepositorySystemSession session, List<ArtifactRepository
         }
     }
 
-    private ArtifactRepositoryLayout getLayout(String id) {
+    private ArtifactRepositoryLayout getLayout(String repoId, String id) {
         ArtifactRepositoryLayout layout = layouts.get(id);
 
+        if (layout == null) {
+            LOGGER.warn("No layout '{}' found for repository id '{}'", id, repoId);
```

Comment:
Won't this warning be printed each time the repository is used?
Should we inform the user about how to fix the problem ?